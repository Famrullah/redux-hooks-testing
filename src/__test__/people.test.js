import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import People from '../pages/people/people';

configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: () => {},
  useSelector: () => ({
    peopleState: {
      people: '',
      isError: '',
      messages: '',
    },
  }),
}));

describe('people component', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<People />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<People />);
    expect(wrapper).toMatchSnapshot();
  });
});
