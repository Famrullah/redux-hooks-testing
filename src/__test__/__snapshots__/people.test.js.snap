// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people component Should render and match the snapshot 1`] = `ShallowWrapper {}`;
